.github/workflows/lint.yml

```diff
@@ -0,0 +1,84 @@
+name: Lint
+on: [push]
+jobs:
+  detect-secrets:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+      - name: Install dependencies
+        run: pip install tox==3.24.4
+      - name: Check for secret lekeage
+        run: tox -e secrets
+  black:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+      - name: Install dependencies
+        run: pip install tox==3.24.4
+      - name: Check that code is formatted with black
+        run: tox -e black
+  isort:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+      - name: Install dependencies
+        run: pip install tox==3.24.4
+      - name: Check that imports are formatted with isort
+        run: tox -e isort
+  flake8:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+      - name: Install dependencies
+        run: pip install tox==3.24.4
+      - name: Lint codebase with flake8
+        run: tox -e flake8
+  bandit:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-python@v4
+        with:
+          python-version: 3.9
+      - name: Install dependencies
+        run: pip install tox==3.24.4
+      - name: Scan codebase for security issues
+        run: tox -e bandit
+  migrations:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Install required ubuntu packages
+        run: sudo apt install libkrb5-dev build-essential python3-dev libldap2-dev libsasl2-dev libssl-dev
```

Comment:
Just out of curiosity, why are these ubuntu packages not needed in the previous steps? These are django specific?

Reply:
Not really specific to Django, but the code needs to be executed in order for these two checks to run, that includes some kerberos and ldap libs that we use, otherwise the installation of the requirements will fail (if you check tox.ini these are the only two that install the entire requirements.txt et al).

The previous ones are all static checks, maybe in the future we can separate them and better take advantage of caching and such, but I was rushing so I could technically have it "done" by the end of the sprint 💃🏽 

I'm also not exactly happy with reusing the Fedora image in `test.yml` but it works for now, so the rest will be future improvements

Reply:
Ah, I see.

You mean using the fedora image inside ubuntu? :smile:

Reply:
yep